### PR TITLE
Port over System.Runtime.InteropServices docs for .NET 9

### PR DESCRIPTION
**Pull request by jkoritzinsky**
## Summary

Fixes https://github.com/dotnet/runtime/issues/105986

### CODE REVIEW

File: xml/System.Runtime.InteropServices/CollectionsMarshal.xml
Comment (carlossanlop) [inline xml/System.Runtime.InteropServices/CollectionsMarshal.xml:301]: Did the porting tool do this? It's supposed to port remarks as plain xml, not as markdown anymore (for simplicity).

```suggestion
        <remarks>
          <para>Items should not be added or removed from the <see cref="T:System.Collections.Generic.Dictionary`2" /> while the ref <paramref name="TValue" /> is in use.</para>
          <para>The ref <see langword="null" /> can be detected by calling <see cref="M:System.Runtime.CompilerServices.Unsafe.IsNullRef``1(``0@)" />.</para>
        </remarks>
```

File: xml/System.Runtime.InteropServices/CollectionsMarshal.xml
Comment (carlossanlop) [inline xml/System.Runtime.InteropServices/CollectionsMarshal.xml:187]: Do you know if `T:System.Collections.Generic.Dictionary`2.AlternateLookup`1` existed in Preview6? Because currently the CI cannot find it. If we introduced it in Preview7 then maybe we will need to merge this after we update dotnet-api-docs to Preview7, to prevent showing something that isn't released yet.

But if it existed in Preview6 already, then I'm not sure what the problem could be. The DocID is well formed, it matches the official DocID: https://github.com/dotnet/dotnet-api-docs/blob/55322a04db5cc185736be7e3cbf56f87cd75ba17/xml/System.Collections.Generic/Dictionary%602%2BAlternateLookup%601.xml#L4

**Review by carlossanlop — APPROVED** [review]
Left one comment for you to consider. There's also a problem with a type that cannot be found. Depending on the root cause, we may have to wait until P7 is released before merging this PR.

File: xml/System.Runtime.InteropServices/CollectionsMarshal.xml
Comment (jkoritzinsky) [reply xml/System.Runtime.InteropServices/CollectionsMarshal.xml:301]: This was copied manually from the above method's docs as I couldn't get the porting tool to work right for CoreLib

File: xml/System.Runtime.InteropServices/CollectionsMarshal.xml
Comment (jkoritzinsky) [reply xml/System.Runtime.InteropServices/CollectionsMarshal.xml:187]: I'm not sure. Honestly CollectionsMarshal isn't in my wheelhouse. I had the wrong format beforehand so lets see if CI likes it after I changed it to match the doc id.

File: xml/System.Runtime.InteropServices/CollectionsMarshal.xml
Comment (carlossanlop) [reply xml/System.Runtime.InteropServices/CollectionsMarshal.xml]: > as I couldn't get the porting tool to work right for CoreLib

Oh, what command did you use? The xml docs for S.P.CoreLib are found under `artifacts\bin\coreclr\<os>.<arch>.<config>\IL\`, that would be the argument for `-IntelliSense`.

File: xml/System.Runtime.InteropServices/CollectionsMarshal.xml
Comment (carlossanlop) [reply xml/System.Runtime.InteropServices/CollectionsMarshal.xml:187]: Alternatively, instead of making these `see crefs` they could just be `<c></c>` for now.

What would be your preference, @gewarren?

File: xml/System.Runtime.InteropServices/CollectionsMarshal.xml
Comment (jkoritzinsky) [reply xml/System.Runtime.InteropServices/CollectionsMarshal.xml]: I did that, but the updated triple slash comments weren't being output into that folder for some reason. I didn't feel like fighting it.

Comment (carlossanlop) [reply xml/System.Runtime.InteropServices/CollectionsMarshal.xml]: Did you also run `build.cmd clr.corelib` before porting?

Comment (jkoritzinsky) [reply xml/System.Runtime.InteropServices/CollectionsMarshal.xml]: Yep. Even deleted my bin artifacts. Probably should have deleted my obj ones as well.

Comment (jkoritzinsky) [reply xml/System.Runtime.InteropServices/CollectionsMarshal.xml]: I had the same problem with System.Runtime.InteropServices initially and solved it by using the obj folder instead. Couldn't get that to work with CoreLib though.

**Pull request merged**
